pytorch_lightning/trainer/trainer.py

```diff
@@ -1494,6 +1500,42 @@ def call_hook(
 
         return output
 
+    def _profile_and_call(self, obj: object, hook_name: str, *args: Any, **kwargs: Any) -> Optional[Any]:
+        fn = getattr(obj, hook_name)
+        with self.profiler.profile(f"{obj.__class__.__name__}.{hook_name}"):
+            return fn(*args, **kwargs)
+
```

Comment:
i'd rather this be directly in the implementation of `call_hook` to avoid making `_profile_and_call` broadly used elsewhere in the trainer codebase. what do you think?

Reply:
Yes, makes sense